RubeesEat/Model/DB/DbPersonRepository.cs

```diff
@@ -0,0 +1,92 @@
+using System.Data;
+
+namespace RubeesEat.Model.DB;
+
+public class DbPersonRepository(IDbConnectionFactory connectionFactory) : IPersonRepository
+{
+    public IReadOnlyList<Person> GetAll()
+    {
+        var persons = new List<Person>();
+
+        using var connection = connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand();
+
+        command.CommandText = "SELECT PersonID, FirstName, LastName FROM Persons ORDER BY LastName ASC;";
+        command.CommandType = CommandType.Text;
+
+        connection.Open();
+
+        using var reader = command.ExecuteReader();
+
+        while (reader.Read())
+        {
+            persons.Add(new Person
+            (
+                reader.GetGuid(0),
+                reader.GetString(1),
+                reader.GetString(2)
+            ));
+        }
+
+        return persons.AsReadOnly();
+    }
+
+    public void Add(Person person)
+    {
+        using var connection = connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand();
+
+        command.CommandText = "INSERT INTO Persons (PersonID, FirstName, LastName) VALUES (@PersonID, @FirstName, @LastName);";
+        command.CommandType = CommandType.Text;
+
+        var personIdParam = command.CreateParameter();
+        personIdParam.ParameterName = "@PersonID";
+        personIdParam.Value = person.Id;
+        command.Parameters.Add(personIdParam);
+
+        var firstNameParam = command.CreateParameter();
+        firstNameParam.ParameterName = "@FirstName";
+        firstNameParam.Value = person.FirstName;
+        command.Parameters.Add(firstNameParam);
+
+        var lastNameParam = command.CreateParameter();
+        lastNameParam.ParameterName = "@LastName";
+        lastNameParam.Value = person.LastName;
+        command.Parameters.Add(lastNameParam);
+
+        connection.Open();
+        command.ExecuteNonQuery();
+    }
+
+
+    public Person GetCurrentUser()
+    {
+        return GetAll()[^1];
+    }
+
+    public Person GetById(Guid guid)
+    {
+        using var connection = connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand();
+
+        command.CommandText = "SELECT PersonID, FirstName, LastName FROM Persons WHERE PersonID = @PersonID;";
+        command.CommandType = CommandType.Text;
+        
+        var personIdParam = command.CreateParameter();
+        personIdParam.ParameterName = "@PersonID";
+        personIdParam.Value = guid;
+        command.Parameters.Add(personIdParam);
+
+        connection.Open();
+
+        using var reader = command.ExecuteReader();
+        reader.Read();
```

Comment:
What happens when `Read` returns `false`?

Reply:
That would mean that there was no person with the searched id. Should we return null in such a case?

Reply:
Yes we should return `null` in such a case.